### PR TITLE
SNET-1151 Get Stake details

### PR DESCRIPTION
**Pull request by rajeevbbqq**

### CODE REVIEW

File: airdrop/application/services/airdrop_services.py [hunk 1]
```diff
@@ -130,14 +123,73 @@ def get_airdrop_window_stake_details(self, inputs):
 
         return status, response
 
-    def get_stake_info(self, staking_contract_address, address):
-        contract = get_contract_instance(
-            STAKING_CONTRACT_PATH, staking_contract_address, contract_name='STAKING')
+    def get_stake_info(self, contract_address, address, airdrop_rewards):
```
Comment (anandrgitnirman) [inline airdrop/application/services/airdrop_services.py:126]: can we call address as user_address here ?

File: airdrop/application/services/airdrop_services.py [hunk 2]
```diff
@@ -130,14 +123,73 @@ def get_airdrop_window_stake_details(self, inputs):
...
+        # User can stake if stake window is open and user can stake
+        is_stakable = True if is_user_can_stake and is_stake_window_is_open else False
+
+        allowed_amount_for_stake = max_stake_amount - already_staked_amount
+
+        if(airdrop_rewards <= allowed_amount_for_stake):
+            # If airdrop rewards is less than allowed amount for stake then stake the full airdrop rewards
+            stakable_amount = airdrop_rewards
+        else:
+            stakable_amount = allowed_amount_for_stake
```
Comment (anandrgitnirman) [inline airdrop/application/services/airdrop_services.py:152]: when airdropw_rewards > allowed amount_for_stake 
=> stakable_amount = airdrop_rewards - allowed_amount_for_stake , Please check this agains the logic in our test case

Comment (anandrgitnirman) [reply airdrop/application/services/airdrop_services.py:152]: please have a test case showing all the scenarios 
-  Full rewaards staked
-  Partial rewards staked
-  full rewards sent back to the user wallet and nothing could be staked as the user exceeded all the limit 
-  full rewards to wallet as the stake window is not open

Comment (rajeevbbqq) [reply airdrop/application/services/airdrop_services.py]: Updated the test case in [commit](https://github.com/singnet/airdrop-services/pull/94/commits/387f89de55b5762ca4b37cfdc8aaef360afee03e)

File: airdrop/application/services/airdrop_services.py (hunk 1 above)
Comment (rajeevbbqq) [reply airdrop/application/services/airdrop_services.py]: Updated the variable to `user_wallet_address` in [commit](https://github.com/singnet/airdrop-services/pull/94/commits/387f89de55b5762ca4b37cfdc8aaef360afee03e)

File: airdrop/application/services/airdrop_services.py [hunk 3]
```diff
@@ -130,14 +123,77 @@ def get_airdrop_window_stake_details(self, inputs):
...
+
+        return is_stakable, stakable_amount, tranfer_to_wallet
+
+    def get_stake_details_of_address(self, contract_address, user_wallet):
+        try:
+            contract = get_contract_instance(
+                STAKING_CONTRACT_PATH, contract_address, contract_name='STAKING')
 
-        return is_stakable, amount
+            stake_info = contract.functions.getStakeInfo(user_wallet).call()
```
Comment (raamb) [inline airdrop/application/services/airdrop_services.py:168]: If user has not staked at all will stake_info still have two elements?

Comment (rajeevbbqq) [reply airdrop/application/services/airdrop_services.py:168]: Yes, It return False & 0 if not staked @raamb

**Review by raamb — APPROVED** [review]

**Pull request merged**